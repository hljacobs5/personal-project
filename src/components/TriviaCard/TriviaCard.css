.buttons {
	display: flex;
    justify-content: space-around;
    flex-direction: column;
    /* margin: auto; */
    /*margin-left: 0%;*/
}

.radio-button {
font-family: 'Kodchasan', sans-serif;
 	color: #6796A3;
}

.questions {
font-family: 'Kodchasan', sans-serif;
 	color: #407888;
}