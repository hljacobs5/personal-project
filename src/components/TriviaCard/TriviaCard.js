import React, { Component } from 'react';
import './TriviaCard.css';
import { connect } from 'react-redux';
import { cleanAnswers } from '../../utilities/helper.js';

class TriviaCard extends Component {
	constructor(props) {
		super();
		this.state = {
			isClicked: ''
		}
	}

	handleChecked = (changeEvent) => {
		this.setState({ isClicked: changeEvent.target.value })
	}

	render() {
		const correctAnswer = this.props.result.correct_answer
		const incorrectAnswers = this.props.result.incorrect_answers
		const answers = cleanAnswers(correctAnswer, incorrectAnswers)
		const buttons = answers.map(answer => {
					return (
						<div>
						   <input type='radio' 
						   		  id={answer.answer} 
						   		  value={answer.answer}
						   		  onClick={this.handleChecked}
						   		  checked={this.state.isClicked === answer.answer} 
						   	/> 
						   <label for={answer.answer}>{answer.answer}</label>
						</div>
					)
				})

		return (
			<div>
				<h2>{this.props.result.question}</h2>
				<div className='buttons'>{buttons}</div>	
			</div>
		)
	  }  
	}



export default TriviaCard;