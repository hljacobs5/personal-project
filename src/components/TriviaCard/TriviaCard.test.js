import React from 'react';
import { shallow, mount } from 'enzyme';
import { TriviaCard, mapStateToProps, mapDispatchToProps } from './TriviaCard';

describe('TriviaCard', () => {
	let wrapper;
	let mockHandleChecked;

	beforeEach(() => {
		mockHandleChecked = jest.fn();
		wrapper = shallow(
			<TriviaCard 
				isClicked={''}
			/>)
	})
	
	it('should exist', () => {
		expect(wrapper).toBeDefined();
	})

	it('should render like snapshot', () => {
		expect(wrapper).toMatchSnapshot();
	})

	// describe('handleChecked', () => {
	// 	it('should update state of isClicked', () => {

	// 	})
	// })
})

