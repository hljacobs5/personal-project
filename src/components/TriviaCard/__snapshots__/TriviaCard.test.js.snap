// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TriviaCard should render like snapshot 1`] = `undefined`;
